sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -69,6 +71,7 @@ class Analyzer(
       WindowsSubstitution ::
       Nil : _*),
     Batch("Resolution", fixedPoint,
+      RewriteFilterSubQuery ::
```

Comment:
I didn't test it if I place it in the last, but should not be a problem, is there any concern for this?